### PR TITLE
New spec for api-version 2017-03-15-preview Version 2

### PR DESCRIPTION
**Pull request by sw47**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by fearthecowboy** [comment]
New API version -- requesting @ravbhatnagar  review.

FYI - it's a simple change -- adding a type to the `definitions` and adding it's usage in `WorkspaceProperties`

``` json
 "Features": {
      "properties": {
        "legacy": {
          "type": "string",
          "description": "The legacy status of the workspace.",
          "enum": [
            "None",
            "GrandfatheredNonSuite",
            "GrandfatheredSuite"
          ],
          "x-ms-enum": {
            "name": "legacyNameEnum",
            "modelAsString": true
          }
        },
        "searchVersion": {
          "type": "string",
          "description": "The search version of the workspace.",
          "enum": [
            "OQL",
            "KQL"
          ],
          "x-ms-enum": {
            "name": "searchVersionNameEnum",
            "modelAsString": true
          }
        },
        "searchVersionUpgradeTime": {
          "type": "string",
          "format": "date-time",
          "description": "The search version upgrade time of the workspace."
        }
      },
      "required": [
        "legacy",
        "searchVersion"
      ],
      "description": "The features of the workspace."
    },
```

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/2017-03-15-preview/OperationalInsights.json [hunk 1]
```diff
@@ -1104,6 +1104,45 @@
...
+          "enum": [
+            "OQL",
+            "KQL"
+          ],
+          "x-ms-enum": {
+            "name": "searchVersionNameEnum",
+            "modelAsString": true
+          }
+        },
+        "searchVersionUpgradeTime": {
```
Comment (ravbhatnagar) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/2017-03-15-preview/OperationalInsights.json]: Is this a readonly property generated by the service? Or something the user needs to provide? It is not clear.

**Review by ravbhatnagar — APPROVED** [review]
Just one minor comment. Else looks fine

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/2017-03-15-preview/OperationalInsights.json [hunk 2]
```diff
@@ -0,0 +1,1320 @@
...
+        },
+        "searchVersion": {
+          "type": "string",
+          "description": "The search version of the workspace.",
+          "enum": [
+            "OQL",
+            "KQL"
+          ],
+          "x-ms-enum": {
+            "name": "searchVersionNameEnum",
```
Comment (fearthecowboy) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/2017-03-15-preview/OperationalInsights.json]: You probably don't need to name this `searchVersionNameEnum` -- `searchVersionName` would be ok, then generated classes don't need the `Enum` suffix in them.

**Review by fearthecowboy — CHANGES_REQUESTED** [review]
@sw47 - just quick points.

**Comment by salameer** [comment]
Hi @sw47,

Please address the provided comments to avoid having this PR closed for inactivity

Thanks,

Samer

**Comment by fearthecowboy** [comment]
Closing PR due to inactivity.

**Comment by smereczynski** [comment]
Can we reopen it and merge?

**Pull request closed without merging**